src/tasks/MonoTargetsTasks/EmitBundleTask/EmitBundleBase.cs

```diff
@@ -0,0 +1,461 @@
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+public abstract class EmitBundleBase : Microsoft.Build.Utilities.Task, ICancelableTask
+{
+    private CancellationTokenSource BuildTaskCancelled { get; } = new();
+
+    private Dictionary<string, string> resourceDataSymbolDictionary = new();
+
+    private Dictionary<string, string[]> resourcesForDataSymbolDictionary= new();
+
+    /// Truncate encoded hashes used in file names and symbols to 24 characters.
+    /// Represents a 128-bit hash output encoded in base64 format.
+    private const int MaxEncodedHashLength = 24;
+
+    /// Must have DestinationFile metadata, which is the output filename
+    /// Could have RegisteredName, otherwise it would be the filename.
+    /// RegisteredName should be prefixed with namespace in form of unix like path. For example: "/usr/share/zoneinfo/"
+    [Required]
+    public ITaskItem[] FilesToBundle { get; set; } = default!;
+
+    /// <summary>
+    /// The function to call before mono runtime initialization
+    /// in order to register the bundled resources in FilesToBundle
+    /// </summary>
+    public string BundleRegistrationFunctionName { get; set; } = "mono_register_resources_bundle";
+
+    /// <summary>
+    /// The filename for the generated source file that registers
+    /// the bundled resources.
+    /// </summary>
+    public string? BundleFile { get; set; }
+
+    /// <summary>
+    /// Path to store build artifacts
+    /// <summary>
+    public string OutputDirectory {get; set; } = default!;
+
+    /// <summary>
+    /// Resources that were bundled
+    ///
+    /// Successful bundling will set the following metadata on the items:
+    ///   - DataSymbol
+    ///   - LenSymbol
+    /// <summary>
+    [Output]
+    public ITaskItem[] BundledResources { get; set; } = default!;
+
+    public override bool Execute()
+    {
+        List<ITaskItem> bundledResources = new(FilesToBundle.Length);
+        foreach (ITaskItem bundledResource in FilesToBundle)
+        {
+            var resourcePath = bundledResource.ItemSpec;
+
+            bundledResource.SetMetadata("ResourceType", "DataResource");
+            try
+            {
+                using FileStream resourceContents = File.OpenRead(resourcePath);
+                using PEReader resourcePEReader = new(resourceContents);
+                if (resourcePEReader.HasMetadata)
+                {
+                    string? managedAssemblyCulture = null;
+
+                    var resourceMetadataReader = PEReaderExtensions.GetMetadataReader(resourcePEReader);
+                    if (resourceMetadataReader.IsAssembly)
```

Comment:
Does this return true for satellite assemblies also?

Reply:
Yes, I believe it does. Put in some logging and saw that this block was hit for satellite assemblies as well